packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -37,139 +24,110 @@ import { useApi } from './hooks/root/useApi';
 import { useGridContainerProps } from './hooks/root/useGridContainerProps';
 import { useEvents } from './hooks/root/useEvents';
 import { useGridKeyboard } from './hooks/features/keyboard/useGridKeyboard';
-import { useErrorHandler } from './hooks/utils/useErrorHandler';
-import { useLogger, useLoggerFactory } from './hooks/utils/useLogger';
+import { useLoggerFactory } from './hooks/utils/useLogger';
 import { useOptionsProp } from './hooks/utils/useOptionsProp';
 import { useRenderInfoLog } from './hooks/utils/useRenderInfoLog';
 import { useResizeContainer } from './hooks/utils/useResizeContainer';
 import { useGridVirtualRows } from './hooks/features/virtualization/useGridVirtualRows';
 import { useGridDensity } from './hooks/features/density';
 import { useStateProp } from './hooks/utils/useStateProp';
-import { GridRootContainerRef } from './models/gridRootContainerRef';
-import { GridApiContext } from './components/GridApiContext';
+import { GridApiContext, GridPropsContext } from './components/GridApiContext';
 import { useGridFilter } from './hooks/features/filter/useGridFilter';
 import { useLocaleText } from './hooks/features/localeText/useLocaleText';
 import { useGridCsvExport } from './hooks/features/export';
 import { useGridInfiniteLoader } from './hooks/features/infiniteLoader';
-import { visibleGridRowCountSelector } from './hooks/features/filter/gridFilterSelector';
 import { useGridFreezeRows } from './hooks/features/rows/useGridFreezeRows';
 
-export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps>(
-  function GridComponent(props, ref) {
-    const rootContainerRef: GridRootContainerRef = React.useRef<HTMLDivElement>(null);
-    const handleRef = useForkRef(rootContainerRef, ref);
+const useXGrid = (apiRef, props) => {
```

Comment:
I was thinking about this as well. I was thinking in a direction of having one hook, like `useCoreGrid` / useBaseGrid` or smth that is basically all the core functionality (the OS part of the grid). The reason is that we are not going to have a case where a community feature is not available for the commercial version. Then for the `XGrid`, we can just list all the remaining commercial features?